cmake/CCCLBuildCompilerTargets.cmake

```diff
@@ -24,6 +24,7 @@ option(CCCL_ENABLE_EXCEPTIONS "Enable exceptions within CCCL libraries." ON)
 option(CCCL_ENABLE_RTTI "Enable RTTI within CCCL libraries." ON)
 option(CCCL_ENABLE_WERROR "Treat warnings as errors for CCCL targets." ON)
 option(CCCL_SUPPRESS_ICC_DEPRECATION_WARNING "Suppress Intel Compiler deprecation warnings" OFF)
+option(CCCL_SUPPRESS_MSVC2017_DEPRECATION_WARNING "Suppress Visual Studio 2017 deprecation warnings" OFF)
```

Comment:
Question: Should we combine them into just `CCCL_SUPPRESS_COMPILER_DEPRECATION_WARNING`

We might collect a bunch of those

Reply:
#2996